components/selection/selection-mixin.js

```diff
@@ -60,6 +60,10 @@ export const SelectionMixin = superclass => class extends RtlMixin(superclass) {
 		this.addEventListener('d2l-selection-select-all-change', this._handleSelectionSelectAllChange);
 		this.addEventListener('d2l-selection-observer-subscribe', this._handleSelectionObserverSubscribe);
 		this.addEventListener('d2l-selection-input-subscribe', this._handleSelectionInputSubscribe);
+
+		const externalSubscribers = this.getRootNode().querySelectorAll(`[subscribed-to="${this.id}"]`);
```

Comment:
Originally I was using a mutation observer similar to the `LabelledMixin`, but it was being added too late in the filter scenario and the selection components had to register themselves anyways.  So what I have currently is a two-way registration system.  If the components try to register themselves but the collection (ex. `list`) they want to subscribe to doesn't exist yet, this code will add them once it is connected.  Likewise, if a component is added after this code has run, [it will subscribe itself](https://github.com/BrightspaceUI/core/pull/1549/files#diff-a7412e30193c0f86945842eaebc7ea7709b8f1b5a541f5b42a7bd56026f5735aR39).

Reply:
Interesting. Where were you adding the mutation observer?

Reply:
I created it here in the `connectedCallback` and had it watching all children of the `this.getRootNode()`.  I tried moving it to the `constructor` and it still seemed to miss it, but I can confirm that again

Reply:
It is possible the subscriber was added before the connectedCallback for this component...?

Reply:
One place I was thinking for the observer was actually in the subscriber. So it would do a `querySelector` for it in `updated()` and if it didn't find it, it could add the mutation observer.

Reply:
Hmm that could work, and then cleans up these lines so the subscriber is completely in charge of subscribing.  Is that "heavier" though... to be listening to all mutations for when the list arrives, rather than have the list tell me when it's arrived?  It means I could handle the case where the list suddenly disappears, which I'm realizing isn't really handled right now.

Reply:
Ideally mutation observer is supposed to be performant, but yeah it could be a bit heavy. You can limit the things you listen for -- so in your case you only care about additions/removals I think. And then yeah it would then suddenly cover the case where the list was removed entirely and you could act accordingly.